src/vec_runtime_DYN.c

```diff
@@ -0,0 +1,210 @@
+ /*
+ Copyright (c) [2019] Steven Munroe.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+
+ vec_runtime_DYN.c
+
+ Contributors:
+      Steven Munroe
+      Created on: Sep 17, 2019
+ */
+
+/*!
+ * \file  vec_runtime_DYN.c
+ * \brief Source file for generating the dynamic IFUNC resolves for
+ * Power Vector Library.
+ *
+ * The working assumption the base and highest supported platform
+ * (-mcpu=) for Big/Little Endian can and should be different.
+ *
+ * The base platform for Big Endian is POWER7 as the first
+ * implementation of Vector Scalar Extended (VXS). The highest
+ * supported platform is POWER8.
+ * Distros supporting Big Endian are older and usual restricted to
+ * POWER8 Compatibility mode when running on POWER9 hardware.
+ *
+ * The base platform for Little Endian is POWER8 as the first
+ * officially supported LE platform. The highest supported platform
+ * is is currently POWER9.  All the latest Distros are Little Endian.
+ *
+ * We depend on the compiler and runtime (GLIBC 2.23) to support
+ * __builtin_cpu_is() in the resolver function.
+ * GCC support for __builtin_cpu_is() on PPC started
+ * with version 6. You need both (GCC 6 or later and GLIBC 2.23 or
+ * later).  We check that the compiler defines the macro
+ * __BUILTIN_CPU_SUPPORTS__ to verify support is available.
+ *
+ * If __BUILTIN_CPU_SUPPORTS__ is not defined we default to the
+ * appropriate (for the platform endian) base platform.
+ */
+
+#include <pveclib/vec_int512_ppc.h>
+
+/*! \brief Macro to expand the parameterize resolver. */
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+#ifdef  __BUILTIN_CPU_SUPPORTS__
+#define VEC_DYN_RESOLVER(FNAME) \
+  if (__builtin_cpu_is ("power8")) \
+    return FNAME ## _PWR8; \
+  else \
+    return FNAME ## _PWR7;
+#else
+#define VEC_DYN_RESOLVER(__FNAME)
+   return FNAME ## _PWR7;
+#endif
+#else // __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#ifdef  __BUILTIN_CPU_SUPPORTS__
+#define VEC_DYN_RESOLVER(FNAME) \
+  if (__builtin_cpu_is ("power9")) \
+    return FNAME ## _PWR9; \
+  else \
+    return FNAME ## _PWR8;
+#else
+#define VEC_DYN_RESOLVER(FNAME) \
+   return FNAME ## _PWR8;
+#endif
+#endif
+
+
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+/* Not BIG Endian, lets call the whole thing off.
+   LITTLE Endian requires power8 or later.  */
```

Comment:
This comment seems out of place, saying "Not BIG Endian", but right under `#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__`

Reply:
Ok will try.

// POWER7supports only BIG Endian. So declare PWR7 externs only for BE.